src/installer/pkg/sfx/Microsoft.NETCore.App/Microsoft.NETCore.App.MonoCrossAOT.sfxproj

```diff
@@ -1,5 +1,12 @@
-<Project Sdk="Microsoft.NET.Sdk">
-  <Sdk Name="Microsoft.DotNet.SharedFramework.Sdk" />
+<Project>
+  <PropertyGroup>
+    <OutDirName Condition="'$(TargetCrossRid)' != ''">monocrossaot-pack.$(TargetCrossRid)</OutDirName>
+    <OutDirName Condition="'$(TargetCrossRid)' == ''">monocrossaot-pack.restore</OutDirName>
+    <RestoreOutputPath>$([System.IO.Path]::GetFullPath('$(ArtifactsObjDir)monocrossaot-pack.restore\'))</RestoreOutputPath>
+    <ProjectAssetsFile>$(RestoreOutputPath)\project.assets.json</ProjectAssetsFile>
```

Comment:
Both these properties shouldn't be required. OutDirName should handle setting the Intermediate and Output paths correctly which these properties then use.

Reply:
We need to set these so all the rid-specific projects get the ridless restore information.

Reply:
Why can't the rid specific projects get restored separately?

Reply:
As we spoke about offline, the rid-specific projects are only differentiated by global properties, which NuGet static graph restore drops. Without this, NuGet only restores the first referenced one and the `project.assets.json` won't be available for other RID instantiations.

Reply:
Oh I see, yes. That's one of the reasons why I had to create separate projects per RID for System.IO.Ports: https://github.com/dotnet/runtime/tree/main/src/libraries/System.IO.Ports/pkg

NuGet pack uses the information from project.assets.json so while the above workaround should work for restore, it doesn't work for pack. But that shouldn't be an issue here I guess.

Please add a comment

Reply:
Yeah it should be fine for pack as we don't want dependencies anyway. I'll add a comment.